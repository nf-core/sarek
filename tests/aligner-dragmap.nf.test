nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_sarek"
    tag "cpu"

    test("-profile test --aligner dragmap --save_reference skip QC/recal/md") {

        when {
            params {
                aligner = 'dragmap'
                outdir = "$outputDir"
                save_reference = true
                skip_tools = 'baserecalibrator,fastqc,markduplicates,mosdepth,multiqc,samtools'
                tools = ''
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // cram_files: All cram files
            def cram_files  = getAllFilesFromDir(params.outdir, include: ['**/*.cram'])
            // Fasta file for cram verification with nft-bam
            def fasta_base  = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            def fasta       = fasta_base + 'genomics/homo_sapiens/genome/genome.fasta'
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_sarek_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All cram files
                    cram_files.collect { file -> [file.getName(), cram(file.toString(), fasta).readsMD5] }
                ).match() }
            )
        }
    }

    test("-profile test --aligner dragmap --save_reference --build_only_index") {

        when {
            params {
                aligner = 'dragmap'
                build_only_index = true
                input = false
                outdir = "$outputDir"
                save_reference = true
                skip_tools = 'multiqc'
                tools = ''
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_sarek_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
