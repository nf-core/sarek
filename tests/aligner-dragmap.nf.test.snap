{
    "Run with profile test | --aligner dragmap --save_reference --build_only_index": {
        "content": [
            5,
            {
                "Workflow": {
                    "nf-core/sarek": "v3.6.0dev"
                }
            },
            [
                "csv",
                "pipeline_info",
                "pipeline_info/nf_core_sarek_software_mqc_versions.yml",
                "reference",
                "reference/dragmap",
                "reference/dragmap/hash_table.cfg",
                "reference/dragmap/hash_table.cfg.bin",
                "reference/dragmap/hash_table.cmp",
                "reference/dragmap/hash_table_stats.txt",
                "reference/dragmap/ref_index.bin",
                "reference/dragmap/reference.bin",
                "reference/dragmap/repeat_mask.bin",
                "reference/dragmap/str_table.bin",
                "reference/intervals",
                "reference/intervals/chr22_1-40001.bed",
                "reference/intervals/chr22_1-40001.bed.gz",
                "reference/intervals/genome.bed",
                "reference/intervals/genome.bed.gz"
            ],
            [
                "hash_table.cmp:md5,1caab4ffc89f81ace615a2e813295cf4",
                "ref_index.bin:md5,dbb5c7d26b974e0ac338024fe4535044",
                "reference.bin:md5,be67b80ee48aa96b383fd72f1ccfefea",
                "repeat_mask.bin:md5,294939f1f80aa7f4a70b9b537e4c0f21",
                "str_table.bin:md5,45f7818c4a10fdeed04db7a34b5f9ff1",
                "chr22_1-40001.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7",
                "chr22_1-40001.bed.gz:md5,87a15eb9c2ff20ccd5cd8735a28708f7",
                "genome.bed:md5,a87dc7d20ebca626f65cc16ff6c97a3e",
                "genome.bed.gz:md5,a87dc7d20ebca626f65cc16ff6c97a3e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.09.0"
        },
        "timestamp": "2024-10-08T11:28:56.135088"
    },
    "Run with profile test | --aligner dragmap --save_reference | skip QC/recal/md": {
        "content": [
            10,
            {
                "BAM_TO_CRAM_MAPPING": {
                    "samtools": 1.21
                },
                "DRAGMAP_ALIGN": {
                    "dragmap": "1.2.1",
                    "samtools": "1.19.2",
                    "pigz": "2.3.4"
                },
                "INDEX_MERGE_BAM": {
                    "samtools": 1.21
                },
                "MERGE_BAM": {
                    "samtools": 1.21
                },
                "Workflow": {
                    "nf-core/sarek": "v3.6.0dev"
                }
            },
            [
                "csv",
                "csv/mapped.csv",
                "pipeline_info",
                "pipeline_info/nf_core_sarek_software_mqc_versions.yml",
                "preprocessing",
                "preprocessing/mapped",
                "preprocessing/mapped/test",
                "preprocessing/mapped/test/test.sorted.cram",
                "preprocessing/mapped/test/test.sorted.cram.crai",
                "reference",
                "reference/dragmap",
                "reference/dragmap/hash_table.cfg",
                "reference/dragmap/hash_table.cfg.bin",
                "reference/dragmap/hash_table.cmp",
                "reference/dragmap/hash_table_stats.txt",
                "reference/dragmap/ref_index.bin",
                "reference/dragmap/reference.bin",
                "reference/dragmap/repeat_mask.bin",
                "reference/dragmap/str_table.bin",
                "reference/intervals",
                "reference/intervals/chr22_1-40001.bed",
                "reference/intervals/chr22_1-40001.bed.gz",
                "reference/intervals/genome.bed",
                "reference/intervals/genome.bed.gz"
            ],
            [
                "hash_table.cmp:md5,1caab4ffc89f81ace615a2e813295cf4",
                "ref_index.bin:md5,dbb5c7d26b974e0ac338024fe4535044",
                "reference.bin:md5,be67b80ee48aa96b383fd72f1ccfefea",
                "repeat_mask.bin:md5,294939f1f80aa7f4a70b9b537e4c0f21",
                "str_table.bin:md5,45f7818c4a10fdeed04db7a34b5f9ff1",
                "chr22_1-40001.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7",
                "chr22_1-40001.bed.gz:md5,87a15eb9c2ff20ccd5cd8735a28708f7",
                "genome.bed:md5,a87dc7d20ebca626f65cc16ff6c97a3e",
                "genome.bed.gz:md5,a87dc7d20ebca626f65cc16ff6c97a3e"
            ],
            [
                [
                    "test.sorted.cram",
                    "7088dc71e5390aec0dd9d778f4568297"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.09.0"
        },
        "timestamp": "2024-10-08T11:28:12.076906"
    }
}
